internal/impl/redpanda/processor_data_transforms.go

```diff
@@ -0,0 +1,414 @@
+// Copyright 2024 Redpanda Data, Inc.
```

Comment:
Minor nit: I'd consider renaming this file to `processor_data_transform.go` and the tests to `processor_data_transform_test.go` since that's the name of the processor.

Reply:
SGTM